### PR TITLE
Added Update to Guava Library

### PR DESCRIPTION
**Pull request by jaimehta19**
## Explanation

This PR updated the Guava library from version `28.1-android` to version `30.1-android` which was relased on December 14th, 2020.

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/misc.xml:9]: Revert these changes

File: .idea/runConfigurations.xml [hunk 2]
```diff
@@ -3,10 +3,11 @@
...
       <set>
+        <option value="com.android.tools.idea.compose.preview.runconfiguration.ComposePreviewRunConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
         <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
       </set>
     </option>
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/runConfigurations.xml:13]: Revert these changes.

File: app/build.gradle [hunk 3]
```diff
@@ -119,7 +119,7 @@ dependencies {
       'com.google.firebase:firebase-analytics-ktx:17.5.0',
       'com.google.firebase:firebase-core:17.5.0',
       'com.google.firebase:firebase-crashlytics:17.0.0',
-      'com.google.guava:guava:28.1-android',
```
Comment (rt4914) [inline app/build.gradle:122]: We need this change in `third_party\versions.bzl` too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@jaimehta1912 PTAL

**Comment by BenHenning** [comment]
Hi @jaimehta1912. Thanks for sending this PR! In general, we usually only update dependency versions when there's a clear need (either a security or privacy update is needed, or we need newer API functionality). Otherwise, updating dependencies can actually reduce the stability of the app since changes in libraries can change app behavior in ways not always covered by our tests.

If you think there's a specific reason why we should update these versions, please follow up with a comment. Otherwise, I'm closing this PR for now.

If you like to help out with open problems, I suggest heading over to our [starter issues list](https://github.com/oppia/oppia-android/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+no%3Aassignee) for unassigned, open issues that we could use help on.

**Comment by jaimehta19** [comment]
@BenHenning  The changes or the improvements introduced with this version are, to detect potential problems for users now so that can plan to use Java 8 language features in our implementation later. A warning log message when running guava-android under a Java 7 VM. This warning is not guaranteed to be logged when running under Java 7, so please don't rely on it as only warning about future problems. If the warning itself causes trouble, can eliminate it by silencing the logger for `com.google.common.base.MoreObjects$ToStringHelper`.

**Comment by jaimehta19** [comment]
@rt4914

**Pull request closed without merging**